src/main/java/com/jfrog/ide/common/ci/BuildDependencyTree.java

```diff
@@ -95,46 +95,50 @@ private void populateArtifacts(DependencyTree artifactsNode, Module module) {
         }
     }
 
-    private void populateDependencies(DependencyTree dependenciesNode, Module module) {
-        Set<Dependency> directDependencies = Sets.newHashSet();
+    private void populateDependencies(DependencyTree dependenciesNode, Module module, Log logger) {
         Multimap<String, Dependency> parentToChildren = HashMultimap.create();
         for (Dependency dependency : module.getDependencies()) {
             String[][] requestedBy = dependency.getRequestedBy();
             if (isEmpty(requestedBy) || isEmpty(requestedBy[0])) {
-                directDependencies.add(dependency);
+                // If there is no "requestedBy" field or the module is the parent of the dependency,
+                // the direct parent is the dependencies node.
```

Comment:
The above two description lines should be before the if statement, and not after.